### PR TITLE
Allow filtering retention table

### PR DESCRIPTION
**Pull request by Kacppian**
## Changes

- Processes filters for retention table on the backend
- Adds filter support on the frontend.(PropertiesFilter)

## Checklist
- [x] All querysets/queries filter by Team (if applicable)
- [x] Backend tests (if applicable)
- [x] Cypress E2E tests (if applicable)

### CODE REVIEW

File: posthog/api/action.py [hunk 1]
```diff
@@ -206,9 +206,9 @@ def _calculate_trends(self, request: request.Request) -> List[Dict[str, Any]]:
     @action(methods=["GET"], detail=False)
     def retention(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
         team = request.user.team_set.get()
-        filter = Filter(request=request)
+        properties = request.GET.get("properties", "{}")
```
Comment (timgl) [inline posthog/api/action.py:209]: I'd be great to add a test for this filtering in test_action.py. There are already some tests for retention in there.

**Review by timgl — COMMENTED** [review]
This is awesome, thanks so much for doing this! One small comment, then we're good to merge :)

File: posthog/api/action.py (hunk 1 above)
Comment (Kacppian) [reply posthog/api/action.py:209]: Sure, I was going to add some frontend tests as well like we have for events.

**Comment by timgl** [comment]
🎉

**Pull request merged**